code/datums/components/cursed_item.dm

```diff
@@ -0,0 +1,29 @@
+/datum/component/cursed_item
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	var/required_trait //trait that will avoid triggering the curse
+	var/item_type //used for the text you get upon triggering the curse
```

Comment:
`///trait that will avoid triggering the curse
var/required_trait
///used for the text you get upon triggering the curse
var/item_type
`